command/resource/apply/apply_test.go

```diff
@@ -61,6 +68,129 @@ func TestResourceApplyCommand(t *testing.T) {
 	}
 }
 
+func readResource(t *testing.T, a *agent.TestAgent, extraArgs []string) string {
+	readUi := cli.NewMockUi()
+	readCmd := read.New(readUi)
+
+	args := []string{
+		"-http-addr=" + a.HTTPAddr(),
+		"-token=root",
+	}
+
+	args = append(extraArgs, args...)
+
+	code := readCmd.Run(args)
+	require.Equal(t, 0, code)
+	require.Empty(t, readUi.ErrorWriter.String())
+	return readUi.OutputWriter.String()
+}
+
+func TestResourceApplyCommand_StdIn(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+
+	t.Parallel()
+
+	a := agent.NewTestAgent(t, ``)
+	defer a.Shutdown()
+	testrpc.WaitForTestAgent(t, a.RPC, "dc1")
+
+	t.Run("hcl", func(t *testing.T) {
+		stdinR, stdinW := io.Pipe()
+
+		ui := cli.NewMockUi()
+		c := New(ui)
+		c.testStdin = stdinR
+
+		stdInput := `ID {
+			Type = gvk("demo.v2.Artist")
+			Name = "korn"
+			Tenancy {
+			  Namespace = "default"
+			  Partition = "default"
+			  PeerName = "local"
+			}
+		  }
+		  
+		  Data {
+			Name = "Korn"
+			Genre = "GENRE_METAL"
+		  }
+		  
+		  Metadata = {
+			"foo" = "bar"
+		  }`
+
+		go func() {
```

Comment:
I'm wondering if this could be racy in the case where the stdin would land before the command is executed 🤔 
Another approach I would think of is to make the command Run in a routine and perform the stdin write after when it actually block, but even then there is a race window 😞